src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -243,7 +269,7 @@ Status CoreWorkerPlasmaStoreProvider::Wait(
       RAY_RETURN_NOT_OK(check_signals_());
     }
   }
-  return Status::OK();
+  return UnblockIfNeeded(raylet_client_, ctx);
```

Comment:
@edoakes it looks to me Wait() wasn't block / unblocking. Was this a bug in the core worker implementation of Wait?